### PR TITLE
Refactor class methods

### LINKED ISSUE #107: Move stat tracker methods to their own class
- Move game methods in stat tracker to Game class and call as class methods
- Do the same for Team and GameTeams
- Remove references to class variables in StatTracker 
- Also move tests from Stat Tracker test to class tests

### PR DESCRIPTION
**Pull request by leahriffell**
**What was done:** 
- Removed helper methods that were still in StatTracker 
- Reorganized StatTracker and StatTracker test to have order match the list given in project requirements 
- Created class methods and tests where appropriate 
  - In these cases, removed method from Helpable and test from StatTracker test 
  - If method was also a stat method, left the test in StatTracker test and also tested in class test 

**Notes:**
- Some methods involved multiple classes and because of this, were left as-in in StatTracker 
- Tried removing @games, @game_teams, and @teams from StatTracker initialize method but ran into quite a few errors so thought it would be better to handle in a separate PR 
- All classes are under 150 lines except for StatTracker (168 lines)
  - Removing section headers, consolidating lines in attr_reader and team_info, removing local vars for exclude would get us down to 150 lines in StatTracker but hopefully y'all will find opportunities for consolidation that I've missed so that we don't have to sacrifice readability 
  
closes #107

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -21,4 +25,77 @@ def initialize(info)
...
+    end
+  end
+
+  def self.all 
+    @@all_games
+  end 
+
+  def self.count
+    @@all_games.count
+  end
```
Comment (AngelaGuardia) [inline lib/game.rb]: This one exists in DataSet so it can be removed from here.

File: lib/team.rb [hunk 2]
```diff
@@ -19,4 +22,18 @@ def initialize(team_collection_data)
...
+    end
+  end
+
+  def self.all 
+    @@all_teams
+  end 
+
+  def self.count
+    @@all_teams.count
+  end
```
Comment (AngelaGuardia) [inline lib/team.rb]: This one exists in DataSet so it can be removed from here.

File: test/game_team_test.rb [hunk 3]
```diff
@@ -53,6 +52,81 @@ def test_it_can_generate_a_season_id
...
+                "16"=>[2, 2, 3, 2, 2, 2, 2],
+                "14"=>[1, 2, 3, 1, 1, 0],
+                "28"=>[0, 2, 3, 2, 3, 0, 3, 3, 6, 2],
+                "54"=>[5, 3, 2, 0, 3, 3],
+                "24"=>[0, 2, 1, 1]
+              }
+
+    assert_equal expected, GameTeam.goals_per_game_per_team
+  end
+end
```
Comment (AngelaGuardia) [inline test/game_team_test.rb]: This end is rogue here - probably from the merge.

File: test/game_test.rb [hunk 4]
```diff
@@ -22,17 +25,65 @@ def test_it_can_read_info
...
+    seasons_and_avg_goals = {
+      "20122013" => 3.70,
+      "20142015" => 3.50,
+      "20172018" => 4.40
+    }
+    assert_equal seasons_and_avg_goals, Game.average_goals_by_season
+  end
+
+  def test_average_goals_per_game
+    assert_equal 3.87, Game.average_goals_per_game
```
Comment (AngelaGuardia) [inline test/game_test.rb:86]: An end is missing here

File: test/game_team_test.rb [hunk 5]
```diff
@@ -53,6 +52,81 @@ def test_it_can_generate_a_season_id
...
+
   def test_it_can_find_all_game_teams_that_meet_a_condition
     GameTeam.create('./data/game_teams_fixture.csv')
```
Comment (AngelaGuardia) [inline test/game_team_test.rb]: Let's delete this line since it is in the setup method. This was the test I wrote!

File: test/game_test.rb [hunk 6]
```diff
@@ -22,17 +25,65 @@ def test_it_can_read_info
...
   def test_it_can_find_a_game
     Game.create('./data/games_fixture.csv')
```
Comment (AngelaGuardia) [inline test/game_test.rb]: Let's delete this here as well. And on line 95!

File: lib/game_team.rb [hunk 7]
```diff
@@ -40,4 +44,76 @@ def self.create(game_team_path)
       game_team.season = game_team.generate_season(game_team.game_id)
     end
   end
-end
+
+  def self.all 
+    @@all_game_teams
+  end
```
Comment (AngelaGuardia) [inline lib/game_team.rb]: This one is a duplicate

File: lib/game.rb [hunk 8]
```diff
@@ -21,4 +25,77 @@ def initialize(info)
...
+
+  def self.create(game_path)
+    CSV.foreach(game_path, headers: true, header_converters: :symbol) do |row|
+      @@all_games << Game.new(row.to_h)
+    end
+  end
+
+  def self.all 
+    @@all_games
+  end
```
Comment (AngelaGuardia) [inline lib/game.rb]: This is a duplicate

File: lib/team.rb [hunk 9]
```diff
@@ -19,4 +22,18 @@ def initialize(team_collection_data)
...
+  def self.create(team_path)
+    CSV.foreach(team_path, headers: true, header_converters: :symbol) do |row|
+      @@all_teams << Team.new(row.to_h)
+    end
+  end
+
+  def self.all 
+    @@all_teams
+  end 
+
```
Comment (AngelaGuardia) [inline lib/team.rb]: This is a duplicate

**Review by AngelaGuardia — CHANGES_REQUESTED** [review]
Thanks so much for doing this major PR @leahriffell! You're a rockstar!!

**Comment by leahriffell** [comment]
Thanks for the feedback, Angela! I just made all of the changes requested.

**Review by hopesgit — APPROVED** [review]

**Review by sarynm12 — APPROVED** [review]
Thank you for doing this, Leah! It looks great!

**Review by AngelaGuardia — APPROVED** [review]

**Pull request merged**